docs/api.rst

```diff
@@ -259,6 +268,65 @@ defaults in the future might change.
 Inside the templates the behaviour can be temporarily changed by using
 the `autoescape` block (see :ref:`autoescape-overrides`).
 
+Note that :func:`~jinja2.select_autoescape` offers also a parameter
+`special_extensions` that allows you to define a custom escape function,
+i.e. if you handling LaTeX files
+(see description of :func:`~jinja.select_autoescape` above for example).
+
+To write your own custom escape function simply return it for the
+wanted template extension. For instance if you are fan of peace in
+the world::
+
+    def escape_to_peace(s):
+        """
+        Your custom escape function. You only have to take care
+        that your escaping is done properly, everything else like
+        preventing multiple escapes and marking the string as safe
+        is done by Jinja itself.
+        """
+        s.replace("war", "peace")
+
+    env = Environment(
+        autoescape=select_autoescape(special_extensions={".world": escape_to_peace}),
+        loader=PackageLoader("mypackage"),
+    )
+
+
+Note that for `.world` files the `{{ var|e }}` and `{{ var | escape }}`
+filters are replaced with the custom escape function.
+
+To mark a string as safe please use the :meth:`Environment.get_markup_class`
+instead of direct :class:`Markup` calls::
+
+
+    template = env.get_template("message_to_the.world")
+    # the content of the template is simply assumed to by
+    """
+    <h1>My Message to the world<h1>
+    {{ my_msg }}
+    I was replied with {{ reply }}
+    """
+    # We know that everything that ends on world will use the special
+    mark_safe = env.get_markup_class(".world")
+    my_msg = mark_safe("Make love not war!")
+    reply = "We want war!"
+    template.render(my_msg=my_msg, reply=reply)
+
+
+.. admonition:: A word of caution
+
+    Be aware that mixing files that using different custom escape
+    functions set by autoescape within of one render command,
+    can lead to unexpected behavior.
+    In general the ``{% include %}`` directive works fine but especially
+    ``{% extends %}`` commands can have unexpected outcomes as main template
+    overwrites the context of the included one.
```

Comment:
Can we just make this throw an error? Possibly with a policy setting to disable it - but I'd rather have a sane/safe default...

Reply:
done, have a look. I mark this resolved so far